corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -128,13 +128,16 @@ hqDefine("data_dictionary/js/data_dictionary", [
         };
 
         self.deprecateProperty = function () {
-            if (!toggles.toggleEnabled('GEOSPATIAL') || !self.isGeoCaseProp()) {
+            if (toggles.toggleEnabled('GEOSPATIAL') && self.isGeoCaseProp) {
+                self.confirmGeospatialDeprecation();
+            } else {
                 self.deprecated(true);
-                return;
             }
+        };
 
+        self.confirmGeospatialDeprecation = function () {
             const $modal = $("#deprecate-geospatial-prop-modal").modal('show');
-            $("#deprecate-geospatial-prop-btn").on('click', function () {
+            $("#deprecate-geospatial-prop-btn").off('click').on('click', function () {
```

Comment:
oh wow, I was not aware of the off function.
Does this remove all bindings or just the one being added by the following on function?

Reply:
In my implementation, this is only removing the 'click' binding. You can however, also remove all event handlers from an element by simply passing no arguments to `.off()`.

Reply:
Rocking, thank you!

Reply:
So, it would remove all click bindings?
In our case we already that its only one for now, but if the button had another click binding anywhere, like through the way modal works, that would also get removed?